schema/deploy/shipping/views.sql

```diff
@@ -2911,4 +2913,139 @@ comment on view shipping.uw_reopening_ehs_reporting_v1 is
+            sample.collected::date as sample_collection_date
+        from all_uw_instances
+        left join warehouse.sample using (encounter_id)
+        where redcap_event_name = 'encounter_arm_1'
+    ),
+
+    uw_individuals as (
+        select
+            individual,
+            max(uw_encounters.encountered) filter (where testing_trigger = '1') as latest_invite_date,
+            max(sample_collection_date) filter (where sample_collection_date is not null) as latest_collection_date
+        from uw_enrollments
+        left join uw_encounters using (individual)
+        group by individual
+    )
+
+    select
+        -- Fall back to metadata from the enrollment in case we don't have encounter data
+        coalesce(uw_encounters.redcap_project_id, uw_enrollments.redcap_project_id) as redcap_project_id,
+        coalesce(uw_encounters.redcap_record_id, uw_enrollments.redcap_record_id) as redcap_record_id,
+        coalesce(uw_encounters.redcap_event_name, uw_enrollments.redcap_event_name) as redcap_event_name,
+        coalesce(uw_encounters.redcap_repeat_instance, uw_enrollments.redcap_repeat_instance) as redcap_repeat_instance,
+        coalesce(uw_encounters.encountered, uw_enrollments.encountered) as encountered,
+        individual,
+        tier,
+        latest_invite_date,
+        latest_collection_date,
+        case
+            when screen_positive and daily_symptoms then 1
+            when screen_positive and daily_exposure_known_pos then 2
+            when screen_positive and daily_exposure then 3
+            when latest_invite_date is null and tier = '1' then 4
+            when latest_invite_date is null and tier in ('2', '3') then 5
+            when surge_selected is not null then 6
+            when tier = '1' then 7
+            when tier in ('2', '3') then 8
+            else null
+        end as priority,
+        case
+            when screen_positive and daily_symptoms then 'symptomatic'
+            when screen_positive and daily_exposure_known_pos then 'exposure_to_known_positive'
+            when screen_positive and daily_exposure then 'gathering_over_10'
+            when latest_invite_date is null and tier = '1' then 'tier_1_baseline'
+            when latest_invite_date is null and tier in ('2', '3') then 'tier_2_and_3_baseline'
+            when surge_selected is not null then surge_selected -- surge_selected is a text field with name of surge event
+            when tier = '1' then 'tier_1_surveillance'
+            when tier in ('2', '3') then 'tier_2_and_3_surveillance'
+            else null
+        end as priority_reason
+    from uw_individuals
+    join uw_enrollments using (individual)
+    left join uw_encounters using (individual)
+    -- Filter to participants who have never been invited to test or their last invite was over 7 days before encounter
+    where (latest_invite_date is null or latest_invite_date < uw_encounters.encountered - interval '7 days')
```

Comment:
I'm thinking about the scenario when a participant registers and never completes the daily attestation. That means that we won't have encounters. You're left joining to uw_encounters, so the where (latest_invite_date is null) part will capture this PT for baseline testing ... the PT has never been invited, so include them in the query.
In this scenario without daily attestations, if the PT enrolled and got baseline testing we'd have latest_invite_date populated.  Would this PT not get included in the priority queue any more because they don't have encounters?

Reply:
Even without daily attestations, we would have an encounter if the PT did a baseline test (from kiosk or swab&send). The `latest_invite_date` would be null so they get included in the queue. We don't want them in the queue if the baseline test was within the last 7 days, so that's why I added the filter based on `latest_collection_date` below. 

Not sure if that makes senses? 🤔

Reply:
Trying an example:
PT enrolls on 1/1/2021. They never do a daily attestation.
They fall into the `where (latest_invite_date is null)` and get baseline testing. Like you say, this will create an encounter.
Encounter gets created for date 1/20/2021. This is where the testing internal instrument gets marked, so 1/20/2021 is the latest_invite_date. They get in to test a couple days later, so Kiosk reg is on 1/22/2021. This now changes the encounter date to 1/22/2021. (Kiosk reg date trumps testing internal instrument date). So, latest_invite_date is now 1/22/2021.

This PT never does daily attestation, so there aren't additional encounters created.
latest_invite_date remains 1/22/2021
most recent encountered remains 1/22/2021

Now, latest_invite_date is not null and latest_invite_date is not 7 days before encountered.
It looks like this PT wouldn't be picked up for surveillance because they don't have any more encounters.

Reply:
Ahhh thank you for the detailed example!! Hmmmm, will have to think through this more...

Reply:
Okay, I _think_ I see a solution for this: 
Instead of trying to bundle both daily attestations and baseline/surveillance into one `select` query, I can separate them into two and combine them with `union all`. So the PT will always have a baseline/surveillance record in the queue even if they don't do daily attestations. Does that make sense?

Reply:
Splitting that into two does make a lot of sense. You could even split into 3 .... baseline/surveillance, daily attestations, and surge,  if not for functionality then for making it easier to see what's going on where.
And the rules *might* be different for the different chunks. If a PT got tested six days ago but was today included in a surge test we might want to include them.

Reply:
See if the latest fixup commit makes sense 🙏